src/parsing/scope.ml

```diff
@@ -66,12 +66,9 @@ type lhs_data =
 
 type mode =
   | M_Term of
-      { mutable m_term_new_metas : problem
-      (** Metavariables generated during scoping. *)
-      ; m_term_meta_of_name : string -> meta option
-      (** Returns the meta with the given name. *)
-      ; m_term_meta_of_key : int -> meta option
-      (** Returns the meta with the give key. *)
```

Comment:
Why do you remove m_term_meta_of_name and m_term_meta_of_key? How do you replace them?

Reply:
Scoped terms do not contain meta variables, they only contain so-called placeholders that are turned into meta-variables during typechecking. This is why I asked https://github.com/Deducteam/lambdapi/discussions/667

Reply:
How in your code a placeholder gets replaced by an already existing meta referred to by the user?

Reply:
As far as I understand, we don't need to do that, except for the "focus" tactic, in which case it's enough to just search for the name of the meta variable, right?

Reply:
A user could use metas in many tactics: refine, apply, rewrite.

Reply:
But we could decide to abandon this feature.

Reply:
For all of these things, it would be nice to see what we lose, and we can deal with placeholders to achieve the same thing (hoping that we can).

Reply:
If you do not start with [] but with the metas of the current typing goals:
https://github.com/Deducteam/lambdapi/blob/93082b2c04c0bb2ecb9dddf33a6c61970154bf7b/src/handle/tactic.ml#L183-L184

Reply:
I was rather speaking at the level of Lambdapi files, can we do everything we want without creating and referencing explicitly meta-variables.

Reply:
Difficult to answer such a question. Users always come up with uses you didn't think about.

Reply:
One way to simulate this is as follows. Imagine that you get two subgoals ?G1 and ?G2 with the same type T, and you need a long proof for proving one or the other. To avoid duplicating such a proof, you need to introduce a lemma before the generation of ?G1 and ?G2 with `have l:T {my long proof}`. Then, for proving ?G1 and ?G2, you can simply do `apply l`.